### PR TITLE
Rustup to rustc 1.40.0-nightly (50f8aadd 2019-11-07)

### PR DESCRIPTION
**Pull request by Manishearth**
changelog: Deprecate [`into_iter_on_array`] lint

r? @phansch @oli-obk

### CODE REVIEW

File: tests/ui/functions.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #![warn(clippy::all)]
-#![allow(dead_code)]
+#![allow(dead_code, improper_ctypes)]
```
Comment (flip1995) [inline tests/ui/functions.rs]: This was already addressed in #4781. Same with `needless_pass_by_value.rs` test.

File: clippy_lints/src/attrs.rs [hunk 2]
```diff
@@ -417,11 +417,14 @@ fn check_attrs(cx: &LateContext<'_, '_>, span: Span, name: Name, attrs: &[Attrib
     }
 
     for attr in attrs {
-        if attr.is_sugared_doc {
-            return;
-        }
+        let attr_item = if let AttrKind::Normal(ref attr) = attr.kind {
+            attr
+        } else {
+            continue;
```
Comment (lzutao) [inline clippy_lints/src/attrs.rs]: the old diff is `return`, this one is `continue`. Which one is correct?

Comment (lzutao) [reply clippy_lints/src/attrs.rs]: (Through I think this one is more meaningful.)

File: clippy_lints/src/doc.rs [hunk 3]
```diff
@@ -247,13 +247,11 @@ pub fn check_attrs<'a>(cx: &LateContext<'_, '_>, valid_idents: &FxHashSet<String
...
 
     for attr in attrs {
-        if attr.is_sugared_doc {
-            if let Some(ref current) = attr.value_str() {
-                let current = current.to_string();
-                let (current, current_spans) = strip_doc_comment_decoration(&current, attr.span);
-                spans.extend_from_slice(&current_spans);
-                doc.push_str(&current);
-            }
+        if let AttrKind::DocComment(ref current) = attr.kind {
```
Comment (lzutao) [inline clippy_lints/src/doc.rs]:
```suggestion
        if let AttrKind::DocComment(ref comment) = attr.kind {
```

File: clippy_lints/src/attrs.rs (hunk 2 above)
Comment (flip1995) [reply clippy_lints/src/attrs.rs]: I think `continue` is better here

**Comment by flip1995** [comment]
I'm currently on the rustup fixes for rust-lang/rust#66180, btw

File: clippy_lints/src/approx_const.rs [hunk 4]
```diff
@@ -62,9 +62,11 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for ApproxConstant {
 
 fn check_lit(cx: &LateContext<'_, '_>, lit: &LitKind, e: &Expr) {
     match *lit {
-        LitKind::Float(s, FloatTy::F32) => check_known_consts(cx, e, s, "f32"),
-        LitKind::Float(s, FloatTy::F64) => check_known_consts(cx, e, s, "f64"),
-        LitKind::FloatUnsuffixed(s) => check_known_consts(cx, e, s, "f{32, 64}"),
+        LitKind::Float(s, LitFloatType::Suffixed(fty)) => match fty {
+            FloatTy::F32 => check_known_consts(cx, e, s, "f32"),
+            FloatTy::F64 => check_known_consts(cx, e, s, "f64"),
```
Comment (mati865) [inline clippy_lints/src/approx_const.rs]: `FloatTy` is imported in line 6.

Comment (flip1995) [reply clippy_lints/src/approx_const.rs]: Yes? :thinking:

Comment (mati865) [reply clippy_lints/src/approx_const.rs]: Disregard it, my brain is out of order today.

**Comment by flip1995** [comment]
Comment to 954eb83:

The new rustc lint covers all the test cases, that were removed in this commit.

File: clippy_lints/src/deprecated_lints.rs [hunk 5]
```diff
@@ -130,3 +130,12 @@ declare_deprecated_lint! {
     pub UNUSED_COLLECT,
     "`collect` has been marked as #[must_use] in rustc and that covers all cases of this lint"
 }
+
+/// **What it does:** Nothing. This lint has been deprecated.
+///
+/// **Deprecation reason:** This lint has been uplifted to rustc and is now called
+/// `array_into_iter`.
+declare_deprecated_lint! {
```
Comment (lzutao) [inline clippy_lints/src/deprecated_lints.rs]: Could you move the doc comments to inner the macro?

**Comment by flip1995** [comment]
@bors r+

Travis passed. Let's get this going.

**Comment by Manishearth** [comment]
@lzutao you're not in the Clippy team so bors will ignore all your comments

**Comment by lzutao** [comment]
It worth a try :p. Sorry but could you undo the treeclosed?

**Comment by Manishearth** [comment]
i thought flip had done it but might as well try again

**Pull request merged**